core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -74,7 +74,7 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
   val pollTask: Cancellable =
     context.system.scheduler.schedule(pollInterval(), pollInterval(), self, Poll)(context.dispatcher)
 
-  var requests = Map.empty[TopicPartition, ActorRef]
+  var requests = Map.empty[ActorRef, RequestMessages]
```

Comment:
I think I understand what you mean. Since they all use the same `groupId` they should use unique `TopicPartition`. Then the question is what to do about it if I receive a `RequestMessages` for topics that are already "taken". 

I see two reasons:
1) wrong usage, user has defined things in the wrong way
2) races when switching topics

I see two ways to handle it
a) reject the request, send back an error message, which will fail the stage
b) accept the request, and discard previous request from other stage (send back empty reply, to unblock it)